src/Revit/RevitHyparTools/Create.Floor.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Autodesk.Revit.DB;
+using Elements.Geometry;
+using ElemGeom = Elements.Geometry;
+
+using ADSK = Autodesk.Revit.DB;
+
+namespace Hypar.Revit
+{
+    public static partial class Create
+    {
+        public static Elements.Floor[] FloorsFromRevitFloor(ADSK.Document doc, ADSK.Floor revitFloor)
+        {
+            var profiles = GetProfilesOfTopFacesOfFloor(doc, revitFloor);
+            var thickness = revitFloor.LookupParameter("Thickness")?.AsDouble();
+
+            var floors = new List<Elements.Floor>();
+            foreach (var profile in profiles)
+            {
+                var zMove = profile.Perimeter.Vertices.Max(v => v.Z);
+                var transform = new ElemGeom.Transform(0, 0, -zMove );
+
+                var zeroedProfile = transform.OfProfile(profile);
+
+                transform.Invert();
+                var floorThickness = thickness.HasValue ? Elements.Units.FeetToMeters(thickness.Value): Elements.Units.FeetToMeters(1);
+                // there appears to be a differen
```

Comment:
Don't leave me hanging.

Reply:
I'm sure the suspense is killing you....
story ending added.